<a href="https://colab.research.google.com/github/Shisiajoy/PYTHON-DATA-ANALYSIS/blob/main/pandas09.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Housing Market

Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

Step 2.
Create 3 differents Series, each of length 100, as follows:

The first a random number from 1 to 4

The second a random number from 1 to 3

The third a random number from 10,000 to 30,000

In [2]:
series1 = pd.Series(np.random.randint(1, 5, size=100))
series2 = pd.Series(np.random.randint(1, 4, size=100))
series3 = pd.Series(np.random.randint(10000, 30001, size=100))

In [3]:
series1.head(20)

0     3
1     2
2     2
3     1
4     3
5     2
6     3
7     1
8     3
9     2
10    1
11    3
12    1
13    2
14    2
15    4
16    4
17    4
18    4
19    1
dtype: int64

Step 3. Let's create a DataFrame by joinning the Series by column

In [7]:
df = pd.DataFrame({'bedrs': series1, 'bathrs': series2, 'price_sqr_meter': series3})
df.head()

bedrs  bathrs  price_sqr_meter
0      3       1            22450
1      2       2            17576
2      2       1            14374
3      1       2            18158
4      3       3            17568

Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [5]:
df.columns = ['bedrs', 'bathrs', 'price_sqr_meter']
df.head()

bedrs  bathrs  price_sqr_meter
0      3       1            22450
1      2       2            17576
2      2       1            14374
3      1       2            18158
4      3       3            17568

Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [11]:
bigcolumn = pd.concat([series1, series2, series3], axis=0)
bigcolumn.head()


0    3
1    2
2    2
3    1
4    3
dtype: int64

Step 6. Oops, it seems it is going only until index 99. Is it true?

In [16]:
print(bigcolumn.tail())
print(f"Is it true that the index goes only up to 99? {bigcolumn.index.max() == 99}")

295    13548
296    13663
297    14909
298    27858
299    27511
dtype: int64
Is it true that the index goes only up to 99? False


Step 7. Reindex the DataFrame so it goes from 0 to 299

In [13]:
bigcolumn.reset_index(drop=True, inplace=True)

# Check the reindexed DataFrame
print(bigcolumn.tail())

295    13548
296    13663
297    14909
298    27858
299    27511
dtype: int64
